### PR TITLE
integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by HansZizold**
**CHANGES**
- The bullet gem was installed and used to solve the N+1 problem.
- Capybara was installed and integration tests were written for each view:
  - User index
  - User show
  - Post index
  - Post show
 
[Detailed Instructions](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md)

This project was developed in pair programming modality by:
- @emmiiorji 
- @HansZizold 

Have a nice review!!!

### CODE REVIEW

File: spec/features/posts/index.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  it 'shows comments and likes counts' do
+    expect(page).to have_content("Comments: #{@post1.comment_counter}, Likes: #{@post1.like_counter}")
+  end
+
+  it 'redirects  to user profile after clicking a post' do
+    click_link(@post1.title)
+    expect(page).to have_selector("input[type=submit][value='Like']")
+  end
```
Comment (phelian23) [inline spec/features/posts/index.html.erb_spec.rb:54]: - [ ] Good job on testing most of the features requested in the projects requirement. However, you didn't add a tests that asserts you can see a section for pagination if there are more posts than fit on the view as requested in the projects requirements. Kindly add that.

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @HansZizold,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter error 👍🏾 
✔️ Good commit messages
✔️ All test passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by HansZizold** [comment]
Dear reviewer, thank you very much for your review and feedback.
We just added a test for the pagination part.

Have a nice review!!!

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @HansZizold, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**